scripts/docker/build.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+set -x
+
+CURDIR=$(dirname $(realpath "$0"))
+WORKDIR=$(realpath ${CURDIR}/../..)
+DOCKER_FILE=$WORKDIR/scripts/docker/opensuse-rust.docker
+
+IMAGE_NAME=opensuse-rust
+CONTAINER_NAME=coconut-build
+
+if ! command -v docker &> /dev/null ; then
+    echo "ERR: docker not found"
+    exit
+fi
+
+IMAGE_ID=$(docker images -q ${IMAGE_NAME})
+
+# Build the container image if it's the first time
+if [ -z ${IMAGE_ID} ] ; then
+    docker build -t ${IMAGE_NAME} --build-arg USER_NAME=${USER} --build-arg USER_ID=$(id -u) -f ${DOCKER_FILE} .
+fi
+
+CONTAINER_ID=$(docker ps -q -f name=${CONTAINER_NAME})
+
+# Create and start the the container if it's not running
+if [ -z ${CONTAINER_ID} ] ; then
+
+    CONTAINER_ID=$(docker ps -q -f name=coconut-build)
+
+    if [ -z ${CONTAINER_ID} ] ; then
+        docker create -it --name=${CONTAINER_NAME} --workdir=${WORKDIR} --user ${USER} --mount type=bind,source=${WORKDIR},target=${WORKDIR} ${IMAGE_NAME} /bin/bash
+    fi
+
+    docker start ${CONTAINER_NAME}
+fi
+
+docker exec -it ${CONTAINER_NAME} /bin/bash -c "source $HOME/.cargo/env && make clean && make"
```

Comment:
What about replacing with just `docker run`:
```suggestion
docker run --rm -it --workdir=${WORKDIR} --user ${USER} --userns keep-id --mount type=bind,source=${WORKDIR},target=${WORKDIR} ${IMAGE_NAME} /bin/bash -c "source $HOME/.cargo/env && make clean && make"
```

I added ` --userns keep-id` to properly work with `podman-docker`, eventually what about using podman directly?

Reply:
I can add your `docker run --rm ` command line as the default action. It would be more in-sync with the idea that containers are usually ephemeral.

Since that command line does not scale in terms of time (the SVSM crates are downloaded and the index is rebuilt on every build), I will add an option `-r` that when set will force the build to reuse an existing container.

Your command line is failing in Docker v20.10.12 though `--userns: invalid user mode`. This might be one of those command line arguments that are not supported in both podman and docker. If we want to support podman, it looks like we would need to extend the script.

Reply:
Thanks, LGTM.

We can add podman support later, adding a parameter.